.github/workflows/azure-podvm-image-build.yml

```diff
@@ -0,0 +1,159 @@
+name: Azure PodVM image build
+
+on:
+  workflow_call:
+    outputs:
+      image-id:
+        description: "The PodVM image id"
+        value: ${{ jobs.build-podvm-image.outputs.image-id }}
+      image-version:
+        description: "The PodVM image version"
+        value: ${{ jobs.build-podvm-image.outputs.image-version }}
+
+  workflow_dispatch:
+    inputs:
+      image-version:
+        type: string
+        description: x.y.z
+      git-ref:
+        type: string
+        default: 'main'
+        description: tag, branch, sha
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  AZURE_PODVM_IMAGE_DEF_NAME: "podvm_image0"
+  CLUSTER_NAME: "e2e-test-${{ github.run_id }}-${{ github.run_attempt }}"
+  COMMUNITY_GALLERY_PREFIX: "/CommunityGalleries/cocopodvm-d0e4f35f-5530-4b9c-8596-112487cdea85"
+  GO_VERSION: "1.20"
+  KATA_AGENT_VERSION: "CC-0.7.0"
+  PODVM_IMAGE_NAME: "peerpod-image-${{ github.run_id }}-${{ github.run_attempt }}"
+  RUST_VERSION: "1.70.0"
+  SSH_USERNAME: "peerpod"
+  VM_SIZE: "Standard_D2as_v5"
+
+jobs:
+  build-podvm-image:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: cloud-api-adaptor/azure/image
+    outputs:
+      image-id: "${{ steps.create-podvm-image.outputs.image-id }}"
+      image-version: "${{ steps.create-podvm-image.outputs.image-version }}"
+    if: github.event.inputs.podvm-image-id == ''
+    steps:
+    - name: Clone cloud-api-adaptor repository
+      uses: actions/checkout@v3
+      with:
+        path: cloud-api-adaptor
+        ref: "${{ inputs.git-ref || 'main' }}"
+
+    - name: Generate image version
+      id: generate-image-version
```

Comment:
A check is needed here to see if the input `image-version` is defined, execute this step only when the input is empty.

Reply:
Also if it is defined this step can also be used to take the input `image-version` to be sent to output `image-version` to comply with existing workflow.